op_crates/fetch/lib.rs

```diff
@@ -156,7 +156,20 @@ where
   let status = res.status();
   let mut res_headers = Vec::new();
   for (key, val) in res.headers().iter() {
-    res_headers.push((key.to_string(), val.to_str().unwrap().to_owned()));
+    let key_string = key.to_string();
+
+    if key_string == "location" {
```

Comment:
`Refresh`, `Cookie` and various pingback headers have the same issue. I wouldn't special case, just do a check along the lines of:
```rust
if val.as_bytes().is_ascii() {
  // use as-is
} else {
  // convert to latin-1
}
```
Tangential: just checking "is it ascii" isn't great from a security perspective. Better would be to allow only "visible" ASCII (i.e., no control characters) but that's probably something for another pull request:
```rust
fn is_visible_ascii(c: u8) {
  c == b'\t' || c >= 32 && c < 127
}
```

Reply:
Thank you for your insight!